yt_dlp/extractor/servus.py

```diff
@@ -60,85 +85,80 @@ class ServusIE(InfoExtractor):
-            'thumbnail': thumbnail,
-            'duration': duration,
-            'timestamp': unified_timestamp(video.get('lastPublished')),
-            'series': series,
+            'title': video.get('title'),
+            'description': self._get_description(video_id) or video.get('description'),
+            'thumbnail': video.get('poster'),
+            'duration': float_or_none(video.get('duration')),
+            'timestamp': unified_timestamp(video.get('currentSunrise')),
+            'series': video.get('label'),
             'season': season,
             'season_number': season_number,
             'episode': episode,
             'episode_number': episode_number,
             'formats': formats,
+            'subtitles': subtitles,
         }
+
+    def _get_description(self, video_id):
+        info = self._download_json(
+            f'https://backend.servustv.com/wp-json/rbmh/v2/media_asset/aa_id/{video_id}?fieldset=page',
+            video_id, fatal=False)
+
+        short_description = unescapeHTML(traverse_obj(info, 'stv_short_description', expected_type=str))
+        long_description = unescapeHTML(traverse_obj(info, 'stv_long_description', expected_type=str, default=''))
+        long_description = long_description.replace('\n\n', '\n')
+
+        if short_description and long_description:
+            return f'{short_description}\n\n{long_description}'
+        return long_description or short_description
+
+    def _report_errors(self, video):
+        playability_errors = traverse_obj(video, ('playabilityErrors', ...))
+        assert isinstance(playability_errors, list)
+        if not playability_errors:
+            raise ExtractorError('No videoUrl and no information about errors')
+
+        if 'FSK_BLOCKED' in playability_errors:
+            details = traverse_obj(video, ('playabilityErrorDetails', 'FSK_BLOCKED'), expected_type=dict)
+            assert isinstance(details, dict) or details is None
+            message = ''
+            if details:
+                if details.get('minEveningHour') is not None:
+                    message += f' from {details["minEveningHour"]:0>2}:00'
+                if details.get('maxMorningHour') is not None:
+                    message += f' to {details["maxMorningHour"]:0>2}:00'
+                if message and details.get('minAge') is not None:
+                    message += f' (Minimum age {details["minAge"]})'
+
+            message = (
+                f'Only available{message}' if message
+                else 'Blocked by FSK with unknown availability')
+
+        elif 'NOT_YET_AVAILABLE' in playability_errors:
+            available_from = traverse_obj(
+                video, ('playabilityErrorDetails', 'NOT_YET_AVAILABLE', 'availableFrom'), 'currentSunrise')
+            message = (
+                f'Only available from {available_from}' if available_from
+                else 'Video not yet available with unknown availability')
```

Comment:
```suggestion
            details = traverse_obj(video, ('playabilityErrorDetails', 'FSK_BLOCKED'), expected_type=dict)
            message = format_field(''.join(
                format_field(details, 'minEveningHour', ' from %02d:00'),
                format_field(details, 'maxMorningHour', ' to %02d:00'),
                format_field(details, 'minAge', ' (Minimum age %d)'),
            ), None, 'Only available%s') or 'Blocked by FSK with unknown availability'

        elif 'NOT_YET_AVAILABLE' in playability_errors:
            message = format_field(
                video, (('playabilityErrorDetails', 'NOT_YET_AVAILABLE', 'availableFrom'), 'currentSunrise'),
                'Only available from %s') or 'Video not yet available with unknown availability'
```
(untested)

Reply:
Thanks, I will test it for you

Reply:
Thanks I will test it for you

Reply:
While testing I also discovered a new error code for the download of https://www.servustv.com/unterhaltung/v/aaxherexvzmlcr5sn5mb/: 
ERROR: [Servus] aaxherexvzmlcr5sn5mb: Video unavailable: DATE_RANGE_UNKNOWN

Bit strange that Servus does not report the available date yet, but there are also no details in the object which we could print from, so the default message is good 😄 
![image](https://user-images.githubusercontent.com/36509607/216846130-562d1f05-01af-40e7-b6ef-e92acb8c87ea.jpeg)

Reply:
Was a bit tedious to test, since it is already past 22:00, so I tested with manual test code:

```python

        self._report_errors({
            'playabilityErrors': ['FSK_BLOCKED'],
            'playabilityErrorDetails': {
                'FSK_BLOCKED': {
                    'minEveningHour': 22,
                    'maxMorningHour': 8,
                    'minAge': 16,
                }
            }
        })
```

`ERROR: [Servus] aa52u39fylr3y4errpiq: Only available from 22:00 to 08:00 (Minimum age 16)`

omitting fields works as before, and when no details are not available we get 

`ERROR: [Servus] aa52u39fylr3y4errpiq: Blocked by FSK with unknown availability`

So LGTM

Reply:
(would actually be nice to be able to add "normal" unit tests, but I did not see any documentation on this)

Reply:
The testing framework is very barebones. Currently it does not allow for these things, if important enough regular unit tests will be added in `test/test_...`. The exact error message is not that important, so would not warrant a unit test in itself imo.

I have thought about a rework on how testing is done, have yet to open an issue though. Any ideas (not only regarding this) would be welcome as a new issue, fresh views would maybe help.

Also you forgot to remove the manual test in the latest commit, so it still calls `self._report_errors()`.

Reply:
> Avoid force pushing

my bad, sorry

Reply:
Regarding the testing framework, I don’t have that much python experience, my background is mostly JUnit and jest

So if I had to come up with something on my own I would wrap the current Tests object in some class called test, and we execute all methods in there that are annotated with @Test or start with test_.

for backwards compatibility we can also check if the test object is in there (and might even keep it around), and dedicated tests we can put into those test methods

but I would first check what common python testing frameworks do, and try to integrate with that. In JUnit there are test extensions, which could be used for backwards compatibility with the TEST object, but not sure if that exists in the python world as well